pkg/apis/core/types.go

```diff
@@ -2505,6 +2505,32 @@ type ContainerStatus struct {
 	// +featureGate=InPlacePodVerticalScaling
 	// +optional
 	Resources *ResourceRequirements
+
+	// User represents user identitiy information of the first process in the container
+	// +featureGate=SupplementalGroupsPolicy
+	// +optional
+	User *ContainerUser
+}
+
+// ContainerUser represents user identity information
+type ContainerUser struct {
+	// Linux holds user identity information of the first process of the containers in Linux.
+	// Note that this field cannot be set when spec.os.name is windows.
+	Linux *LinuxContainerUser
```

Comment:
I don't love spelling out Linux here - is there precedent?

Reply:
@thockin sorry for long delay.

> is there precedent?

As far as I know, probably `seLinuxOptions` in `SecurityContext`?

However, I'm not a Windows specialist, though, UID/GID is a specific concept in Linux, isn't it? Then, I don't have a good idea to express this information in the container status.

Would you have suggestions?

Reply:
@thockin I keep this as is because I still have no idea how to avoid spelling out Linux here because supplementalGroups is the idea which is specific to Linux🙇 Any suggestions??

Reply:
@liggitt @tallclair for consult.  I hate that the API spells out "Linux" but the structure would be different for Windows.

Reply:
It is ugly, but I agree the shape for linux and windows is usually different.

For prior art, we have `securityContext.windowsOptions` at the pod and container level. In the cri-api, we also have 

```
    // Optional configurations specific to Linux hosts.
    LinuxPodSandboxConfig linux = 8;
    // Optional configurations specific to Windows hosts.
    WindowsPodSandboxConfig windows = 9;
```

For OS-specific fields like this, do validate that this is not set if `spec.podOS.name` is set to something other than "linux"

Reply:
ACK